modules/instrument_manager/jsx/uploadForm.js

```diff
@@ -0,0 +1,97 @@
+import React, {Component} from 'react';
+import PropTypes from 'prop-types';
+
+class InstrumentUploadForm extends Component {
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      data: {},
+      selectedFile: null,
+    };
+
+    this.fileSelected = this.fileSelected.bind(this);
+    this.upload = this.upload.bind(this);
+  }
+
+  fileSelected(element, file) {
+    this.setState({
+      selectedFile: file,
+    });
+  }
+
+  upload() {
+    const data = new FormData();
+    data.append('install_file', this.state.selectedFile);
+
+    fetch(this.props.action, {
+      method: 'POST',
+      credentials: 'same-origin',
+      body: data,
+    })
+    .then((resp) => {
+      if (resp.status == 201) {
+        swal({
+          title: 'Installation Successful!',
+          type: 'success',
+        }, function() {
+          window.location.assign(loris.BaseURL + '/instrument_manager/');
+        });
+      }
+      return resp.json();
+    })
+    .then((data) => {
+      if (data.message) {
+         swal({
+          title: 'Upload Successful!',
+          type: 'success',
+          text: data.message,
+        }, function() {
+          window.location.assign(loris.BaseURL + '/instrument_manager/');
+        });
+      }
+      if (data.error) {
+         swal({
+          title: 'An error occured',
+          type: 'error',
+          text: data.error,
+        });
+      }
+    })
+    .catch((error) => {
+      this.setState({error: true});
+      console.error(error);
+    });
+  }
+
+  render() {
+    const disabled = () => this.state.selectedFile == null;
```

Comment:
Can you use strict equality here? ie. `===`?

Reply:
yes, the default is null so it wont be undefined.